numba/core/types/misc.py

```diff
@@ -141,17 +141,25 @@ def key(self):
 class CPointer(Type):
     """
     Type class for pointers to other types.
+
+    Attributes:
+        dtype : The type current pointer represents.
+        addrspace (int): The address space current pointer belongs to.
     """
     mutable = True
 
-    def __init__(self, dtype):
+    def __init__(self, dtype, addrspace=None):
         self.dtype = dtype
-        name = "%s*" % dtype
+        self.addrspace = addrspace
+        if addrspace != None:
+            name = "%s_%s*" % (dtype, addrspace)
+        else:
+            name = "%s*" % dtype
```

Comment:
Can you add a simple test so that both branches are exercised?
It can go in https://github.com/numba/numba/blob/5f7bcce048b9c0a7fd02376ca8a2e3574625c99f/numba/tests/test_types.py#L47

Reply:
Test added in the placed in the suggested location.